### DISCUSSION TITLE
Can trivy be used to scan a host operating system ?

### DISCUSSION THREAD

**Discussion by liorkesos** [aquasecurity/trivy#2615] [Q&A]
I'm looking to collect sbom and security information from the host operating system (in this case ubuntu) and several docker-compose installed on it.
The docker format is straightforward but I'd like to use the same tooling to collect sbom get security info or misconfigured from the host operating system as well...

**Comment by knqyf263** [comment]
Yes, you can use the `rootfs` subcommand. We may merge `rootfs` into `fs` in the near future, but `rootfs` is available now.
https://aquasecurity.github.io/trivy/v0.30.4/docs/vulnerability/scanning/rootfs/

```
$ trivy rootfs --format cyclonedx --output host.cdx.json /
```

**Reply by liorkesos** [reply]
Running the command on rootfs actually choked after 5 minutes..
![image](https://user-images.githubusercontent.com/245027/181462174-5ded2313-063e-4979-813d-265d3e4dc856.png)

**Reply by knqyf263** [reply]
Please see the log message.

**Reply by knqyf263** [reply]
If your image is large, it take a while to scan. You can also use `--skip-dirs`.
https://aquasecurity.github.io/trivy/v0.30.4/docs/vulnerability/examples/others/#skip-directories

**Reply by trevor-vaughan** [reply]
@knqyf263 If you merge `fs` and `rootfs` we REALLY need the ability to ignore dev dependencies as a flag. See #1350

**Reply by knqyf263** [reply]
We welcome contribution.

**Reply by trevor-vaughan** [reply]
@knqyf263 This was more of a "please don't break X" based on your feature merge statement. One of the reasons I currently use `trivy` over other solutions is that I don't have to roll a container image.

That said, I'm hoping to get time to contribute to fix some of these things in the future.
